src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1206,6 +1198,22 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       cmdline.get_values("nondet-static-exclude").begin(),
       cmdline.get_values("nondet-static-exclude").end());
 
+    if(
+      cmdline.isset(FLAG_LOOP_CONTRACTS) &&
+      cmdline.isset(FLAG_LOOP_CONTRACTS_NO_UNWIND))
+    {
+      // When the model is produced by Kani, we must not automatically unwind
+      // the backjump introduced by the loop transformation.
+      // Automatic unwinding duplicates assertions found in the loop body, and
+      // since Kani expects property identifiers to remain unique. Having
+      // duplicate instances of the assertions makes Kani fail to handle the
+      // analysis results.
+      log.warning() << "**** WARNING: transformed loops will not be unwound "
+                    << "after applying loop contracts. Remember to unwind "
+                    << "them at least twice to pass unwinding-assertions."
+                    << messaget::eom;
+    }
```

Comment:
Do we have a GitHub issue to track this?

Reply:
It is not really an issue, it is just two modes of operation.